code/game/gamemodes/clock_cult/clock_scripture.dm

```diff
@@ -0,0 +1,1010 @@
+/*
+Tiers and Requirements
+
+Pieces of scripture require certain follower counts, contruction value, and active caches in order to recite.
+Drivers: Unlocked by default
+Scripts: 5 servants and a cache
+Applications: 8 servants, 3 caches, and 50 CV
+Revenant: 10 servants and 100 CV
+Judgement: 10 servants, 100 CV, and any existing AIs are converted or destroyed
+*/
+
+/datum/clockwork_scripture
+	var/name = "scripture"
+	var/desc = "Ancient Ratvarian lore. This piece seems particularly mundane."
+	var/list/invocations = list() //Spoken over time in the ancient language of Ratvar. See clock_unsorted.dm for more details on the language and how to make it.
+	var/channel_time = 10 //In deciseconds, how long a ritual takes to chant
+	var/list/required_components = list("belligerent_eye" = 0, "vanguard_cogwheel" = 0, "guvax_capacitor" = 0, "replicant_alloy" = 0, "hierophant_ansible" = 0) //Components required
+	var/list/consumed_components = list("belligerent_eye" = 0, "vanguard_cogwheel" = 0, "guvax_capacitor" = 0, "replicant_alloy" = 0, "hierophant_ansible" = 0) //Components consumed
+	var/obj/item/clockwork/slab/slab //The parent clockwork slab
+	var/mob/living/invoker //The slab's holder
+	var/whispered = FALSE //If the invocation is whispered rather than spoken aloud
+	var/consumed_component_override = FALSE //If consumed components are unique to a scripture regardless of tier
+	var/usage_tip = "This piece seems to serve no purpose and is a waste of components." //A generalized tip that gives advice on a certain scripture
+	var/invokers_required = 1 //How many people are required, assuming that a scripture requires multiple
+	var/multiple_invokers_used = FALSE //If scripture requires more than one invoker
+	var/multiple_invokers_optional = FALSE //If scripture can have multiple invokers to bolster its effects
+	var/tier = SCRIPTURE_PERIPHERAL //The scripture's tier
+
+/datum/clockwork_scripture/New()
+	..()
+	if(tier < SCRIPTURE_DRIVER && !consumed_component_override) //Anything above a driver consumes components
+		consumed_components = required_components
+
+/datum/clockwork_scripture/proc/run_scripture()
+	if(can_recite() && check_special_requirements())
+		slab.busy = "Invocation ([name]) in progress"
+		if(recital())
+			slab.busy = null
+			if(scripture_effects() && (!ratvar_awakens && !slab.no_cost))
+				for(var/i in required_components)
+					if(tier <= SCRIPTURE_DRIVER || consumed_component_override)
+						if(clockwork_component_cache[i] >= consumed_components[i]) //Draw components from the global cache first
```

Comment:
tbh PJB is right here, just set the consumed and required components for each instead of this